src/scripts/ecr_login.sh

```diff
@@ -3,26 +3,57 @@ AWS_ECR_EVAL_REGION="$(eval echo "${AWS_ECR_STR_REGION}")"
 AWS_ECR_EVAL_PROFILE_NAME="$(eval echo "${AWS_ECR_STR_PROFILE_NAME}")"
 AWS_ECR_EVAL_ACCOUNT_ID="$(eval echo "${AWS_ECR_STR_ACCOUNT_ID}")"
 AWS_ECR_VAL_ACCOUNT_URL="${AWS_ECR_EVAL_ACCOUNT_ID}.dkr.ecr.${AWS_ECR_EVAL_REGION}.${AWS_ECR_STR_AWS_DOMAIN}"
+AWS_ECR_EVAL_PUBLIC_REGISTRY_ALIAS="$(eval echo "${AWS_ECR_STR_PUBLIC_REGISTRY_ALIAS}")"
 ECR_COMMAND="ecr"
 
+eval "$SCRIPT_UTILS"
+detect_os
+set_sudo
+
 if [ -z "${AWS_ECR_EVAL_ACCOUNT_ID}" ]; then
   echo "The account ID is not found. Please add the account ID before continuing."
   exit 1
 fi
 
 if [ "$AWS_ECR_BOOL_PUBLIC_REGISTRY" == "1" ]; then
     AWS_ECR_EVAL_REGION="us-east-1"
-    AWS_ECR_VAL_ACCOUNT_URL="public.ecr.aws"
+    AWS_ECR_VAL_ACCOUNT_URL="public.ecr.aws/${AWS_ECR_EVAL_PUBLIC_REGISTRY_ALIAS}"
     ECR_COMMAND="ecr-public"
-fi
-
-if [ -n "${AWS_ECR_EVAL_PROFILE_NAME}" ]; then
-    set -- "$@" --profile "${AWS_ECR_EVAL_PROFILE_NAME}"
+    aws "${ECR_COMMAND}" get-login-password --region "${AWS_ECR_EVAL_REGION}" --profile "${AWS_ECR_EVAL_PROFILE_NAME}" \
+     | docker login --username AWS --password-stdin "${AWS_ECR_VAL_ACCOUNT_URL}"
+    exit 0
 fi
 
 if [ -f "$HOME/.docker/config.json" ] && grep "${AWS_ECR_VAL_ACCOUNT_URL}" < ~/.docker/config.json > /dev/null 2>&1 ; then
     echo "Credential helper is already installed"
-else
-    docker logout "${AWS_ECR_VAL_ACCOUNT_URL}"
-    aws "${ECR_COMMAND}" get-login-password --region "${AWS_ECR_EVAL_REGION}" "$@" | docker login --username AWS --password-stdin "${AWS_ECR_VAL_ACCOUNT_URL}"
 fi
+
+Configure_config.json(){
```

Comment:
highly recommend not putting `.json` in function names. recommend choosing all lowercase or all upper case as well.

`CONFIGURE_CONFIG_JSON()`

Reply:
is the `config.json` specifically for the ECR credential helper(s)? Or do you see this function being used to configure things more broadly?

I ask because the name could likely be tightened up to match the intent, something like `CONFIGURE_CREDENTIAL_HELPERS()` or something along those lines.

Reply:
Also, it looks like the function as coded will append a new top-level object to a `config.json` that already exists (this will result in invalid json) - you might need to think about using a utility like `jq` to merge the `"credHelpers"` key into the main config if one already exists.

Reply:
@KyleTryon - I've made all of the functions lower cased

@bjohnso5

> is the config.json specifically for the ECR credential helper(s)? Or do you see this function being used to configure things more broadly?

The function strictly updates `config.json` and is only used in the `ecr_login` command.  

> Also, it looks like the function as coded will append a new top-level object to a config.json that already exists (this will result in invalid json) - you might need to think about using a utility like jq to merge the "credHelpers" key into the main config if one already exists.

Good call here. I've added some logic that will append an exisiting json file using `jq`. My only concern is that some executors may not have `jq` installed. I've added a note to the command's description

Reply:
@brivu in that case I'd still recommend a more specific name, like `configure_ecr_login()` or something along those lines